sdk-lib/src/main/java/cash/z/ecc/android/sdk/model/Account.kt

```diff
@@ -1,12 +1,113 @@
+ * the following properties:
+ *
+ * - When an account starts being tracked within a `Synchronizer` instance, a new
+ *   `Account` is generated.
+ * - If an `Account` is present within a `Synchronizer`, it always points to the same
+ *   account.
+ *
+ * What this means is that account identifiers are not stable across "wallet recreation
+ * events". Examples of these include:
+ * - Restoring a wallet from a backed-up seed.
+ * - Importing the same viewing key into two different wallet instances.
  *
- * @param value A 0-based account index.  Must be >= 0.
+ * @param accountUuid The account identifier. Must be length 16.
+ * @param ufvk The account's Unified Full Viewing Key, if any.
+ * @param accountName A human-readable name for the account
+ * @param keySource A string identifier or other metadata describing the source of the seed
+ * @param seedFingerprint The seed fingerprint. Must be length 16.
+ * @param hdAccountIndex ZIP 32 account index
  */
-data class Account(val value: Int) {
+data class Account internal constructor(
+    val accountUuid: ByteArray,
+    val ufvk: String?,
+    val accountName: String?,
+    val keySource: String?,
+    val seedFingerprint: ByteArray?,
+    val hdAccountIndex: Long?,
+) {
     init {
-        require(value >= 0) { "Account index must be >= 0 but actually is $value" }
+        require(accountUuid.size == JNI_ACCOUNT_UUID_BYTES_SIZE) {
+            "Account UUID must be 16 bytes"
+        }
+        seedFingerprint?.let {
+            require(seedFingerprint.size == JNI_ACCOUNT_SEED_FP_BYTES_SIZE) {
+                "Seed fingerprint must be 32 bytes"
+            }
+        }
+    }
+
+    companion object {
+        fun new(jniAccount: JniAccount): Account =
+            Account(
+                accountUuid = jniAccount.accountUuid,
+                ufvk = jniAccount.ufvk,
+                accountName = jniAccount.accountName,
+                keySource = jniAccount.keySource,
+                seedFingerprint = jniAccount.seedFingerprint,
+                // We use -1L to represent NULL across JNI
+                hdAccountIndex = if (jniAccount.hdAccountIndex == -1L) null else jniAccount.hdAccountIndex
+            )
+
+        fun new(accountUuid: ByteArray): Account =
+            Account(
+                accountUuid = accountUuid,
+                ufvk = null,
+                accountName = null,
+                keySource = null,
+                seedFingerprint = null,
+                hdAccountIndex = null
+            )
+    }
+
+    override fun toString(): String {
+        return "Account(accountUuid=${accountUuid.contentToString()}," +
+            " ufvk length=${ufvk?.length}," +
+            " accountName=$accountName," +
+            " keySource=$keySource," +
+            " seedFingerprint size=${seedFingerprint?.size}," +
+            " hdAccountIndex=$hdAccountIndex)"
+    }
+
+    override fun equals(other: Any?): Boolean {
```

Comment:
we probably do not need to override equals and hasCode here

Reply:
The `equals` and `hashCode` are required due to the `ByteArray` parameters in this class.